src/m1_facilitator/models/ERC20GatewayTokenPair.ts

```diff
@@ -0,0 +1,99 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+
+import Comparable from '../../m0_facilitator/observer/Comparable';
+
+import assert = require('assert');
+
+/**
+ * ERC20GatewayTokenPair class implements erc20GatewayTokenPair model to be used
+ * from other parts of implementation.
+ */
+export default class ERC20GatewayTokenPair extends Comparable<ERC20GatewayTokenPair> {
+  /** ERC20Gateway address. */
+  public readonly erc20Gateway: string;
+
+  /** Value token address. */
+  public readonly valueToken: string;
+
+  /** Utility token address. */
+  public readonly utilityToken: string;
+
+  /** Specifies the creation date of the model. */
+  public readonly createdAt?: Date;
+
+  /** Specifies the update date of the model. */
+  public readonly updatedAt?: Date;
+
+  /**
+   * Constructs an ERC20GatewayTokenPair model from the given parameter.
+   *
+   * @param erc20Gateway Newly created erc20GatewayTokenPair's erc20 gateway address.
+   * @param valueToken Newly created erc20GatewayTokenPair's value token address.
+   * @param utilityToken Newly created erc20GatewayTokenPair's utility token address.
+   * @param [createdAt] Specifies the model creation date.
+   * @param [updatedAt] Specifies the model update date.
+   *
+   * @pre erc20Gateway is not an empty string.
+   * @pre valueToken is not an empty string.
+   * @pre utilityToken is not an empty string.
+   */
+  public constructor(
+    erc20Gateway: string,
+    valueToken: string,
+    utilityToken: string,
+    createdAt?: Date,
+    updatedAt?: Date,
+  ) {
+    super();
+
+    assert(erc20Gateway !== '');
+    assert(valueToken !== '');
+    assert(utilityToken !== '');
+
+    this.erc20Gateway = erc20Gateway;
+    this.valueToken = valueToken;
+    this.utilityToken = utilityToken;
+
+    this.createdAt = createdAt;
+    this.updatedAt = updatedAt;
+  }
+
+  /**
+   * compareTo() function implements comparison between two erc20GatewayTokenPair objects.
+   *
+   * The function first compares case-insensitively the `erc20Gateway` attributes
+   * and if the attributes for the both objects are equal, compares case-insensitively
+   * `valueToken` attributes.
+   *
+   * @param other A erc20GatewayTokenPair object to compare with.
+   *
+   * @returns `> 0` if the current object is greater than the given one.
+   *          `0` if the current object is equal to given one.
+   *          `< 0` if the current object is less than the given one.
+   */
+  public compareTo(other: ERC20GatewayTokenPair): number {
+    const erc20GatewaysComparison: number = this.erc20Gateway.localeCompare(
```

Comment:
This can be simply compared by concatening "{erc20Gateway}-{valueToken}" pair. Code will get much simpler.
```
const currentKey = this. erc20Gateway.concat(this. valueToken);
const specifiedKey = other.erc20Gateway.concat(other.erc20Gateway);
return currentKey.localeCompare(specifiedKey, 'en', { sensitivity: 'base' });
```

Reply:
The existing implementation is cleaner for me. I will propose to keep that one.

Reply:
You can keep your proposal but I still feel above suggested approach is very clean.